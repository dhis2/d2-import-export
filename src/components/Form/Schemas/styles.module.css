.container {
    display: flex;
    flex-direction: column;
}
.container > div:nth-child(1) > div {
    margin-right: 15px;
}
form .formControl {
    margin-top: 10px;
    display: flex;
    flex-direction: column;
    height: 1100px;
    flex-wrap: wrap;
}

form .formLabel {
    margin-bottom: 5px;
    width: 100%;
    user-select: none;
    color: #000;
    font-size: 13px;
    font-weight: 600;
    text-transform: uppercase;
}

.group {
    width: 30%;
    margin-bottom: 10px;
}

.group label {
    font-size: 14px;
}

.group .schema {
    margin-top: 5px;
}

.group .schema label {
    height: 24px;
    width: 33%;
    margin-right: 0;
    white-space: nowrap;
}
.group .schema > div {
    width: auto;
    font-size: 14px;
    align-items: flex-start;
    line-height: inherit;
}

.group .schema > div label {
    margin-left: 4px;
}

.controls {
    display: flex;
    flex-direction: row;
    margin: 15px 0;
    width: 100%;
}
