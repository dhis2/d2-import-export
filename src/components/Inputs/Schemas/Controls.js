import { useForm } from 'react-final-form'
import { Button } from '@dhis2/ui-core'
import React, { useCallback } from 'react'
import i18n from '@dhis2/d2-i18n'
import s from '../Schemas.module.css'

const ToggleAll = ({ schemaKey, value, children, className }) => {
    const form = useForm()

    const onClick = useCallback(() => {
        const fieldKeys = Object.keys(form.getState().values[schemaKey])

        form.batch(() => {
            fieldKeys.forEach(fieldKey => {
                form.change(`${schemaKey}.${fieldKey}`, value)
            })
        })
    }, [form, value, schemaKey])

    return (
        <Button
            onClick={onClick}
            type="button"
            children={children}
            className={className}
        />
    )
}

const SelectAll = ({ schemaKey }) => (
    <ToggleAll schemaKey={schemaKey} value={true} className={s.selectAllButton}>
        {i18n.t('Select All')}
    </ToggleAll>
)

const SelectNone = ({ schemaKey }) => (
    <ToggleAll schemaKey={schemaKey} value={false}>
        {i18n.t('Select None')}
    </ToggleAll>
)

export const Controls = ({ schemaKey }) => (
    <div className={s.controls}>
        <SelectAll schemaKey={schemaKey} />
        <SelectNone schemaKey={schemaKey} />
    </div>
)
