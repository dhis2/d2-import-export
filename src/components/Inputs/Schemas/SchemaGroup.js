import React from 'react'
import s from '../Schemas.module.css'
import { Checkbox } from '../../FinalFormComponents/Checkbox'

export const SchemaGroup = ({ label, schemas }) => (
    <div className={s.group}>
        <span className={s.formLabel}>{label}</span>
        <div className={s.schema}>
            {schemas.map(schema => (
                <div key={schema.name}>
                    <Checkbox
                        label={schema.label}
                        name={`schemas.${schema.name}`}
                        format={v => v.toString()}
                        parse={v => Boolean(v)}
                        initialValue="true"
                    />
                </div>
            ))}
        </div>
    </div>
)
