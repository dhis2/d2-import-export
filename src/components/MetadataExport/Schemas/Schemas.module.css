.container {
    display: flex;
    flex-direction: column;
}

.container > div:nth-child(1) > div {
    margin-right: 15px;
}

.group {
    width: 30%;
    margin-bottom: 10px;
}

.group .schema {
    margin-top: 5px;
}

.formLabel {
    margin-bottom: 5px;
    width: 100%;
    user-select: none;
    color: #000;
    font-size: 13px;
    font-weight: 600;
    text-transform: uppercase;
}

.formControl {
    margin-top: 10px;
    display: flex;
    flex-direction: column;
    max-height: 1100px;
    flex-wrap: wrap;
}
