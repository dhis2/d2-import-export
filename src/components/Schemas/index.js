export { Schemas } from './Schemas'
export { SchemasField, SINGLE_SCHEMA_VALIDATOR } from './SchemasField'
