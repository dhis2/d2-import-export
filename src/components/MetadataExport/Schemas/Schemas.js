import React, { useEffect, useState } from 'react';
import { useDataQuery } from '@dhis2/app-runtime';
import i18n from '@dhis2/d2-i18n';
import { Button, ButtonStrip, CircularLoader } from '@dhis2/ui-core';

import {
    filterOutExcludedSchemas,
    groupName,
    formatSchemas,
    getSchemaGroups,
    getGroupLabels,
    getGroupOrder,
} from './helper';
import { SchemaGroup } from './SchemaGroup';
import s from './Schemas.module.css';

const schemaQuery = {
    schemas: {
        resource: 'schemas.json',
    },
};

const Schemas = ({ excludeSchemas, setCheckedSchemas, checkedByDefault }) => {
    const { loading, error, data } = useDataQuery(schemaQuery);
    const [schemaGroups, setSchemaGroups] = useState(undefined);
    const [schemaGroupLabels, setSchemaGroupLabels] = useState(undefined);
    const [schemaGroupOrder, setSchemaGroupOrder] = useState(undefined);

    useEffect(() => {
        if (data) {
            const schemas = data.schemas.schemas;
            const filteredSchemas = filterOutExcludedSchemas(
                excludeSchemas,
                schemas
            );
            const formattedSchemas = formatSchemas(
                filteredSchemas,
                checkedByDefault
            );
            const groups = getSchemaGroups(formattedSchemas);
            setSchemaGroups(groups);
            setSchemaGroupOrder(getGroupOrder(groups));
            setSchemaGroupLabels(getGroupLabels(groups));
            propagateCheckedSchemas(groups);
        }
    }, [data]);

    const propagateCheckedSchemas = updatedSchemaGroups => {
        setCheckedSchemas(
            Object.keys(updatedSchemaGroups).reduce(
                (acc, groupName) => [
                    ...acc,
                    ...updatedSchemaGroups[groupName]
                        .filter(s => s.checked)
                        .map(s => s.name),
                ],
                []
            )
        );
    };

    const selectGenericHandler = val => () => {
        const updatedSchemaGroups = Object.keys(schemaGroups).reduce(
            (acc, groupName) => ({
                ...acc,
                [groupName]: schemaGroups[groupName].reduce(
                    (grAcc, group) => [...grAcc, { ...group, checked: val }],
                    []
                ),
            }),
            {}
        );
        setSchemaGroups(updatedSchemaGroups);
        propagateCheckedSchemas(updatedSchemaGroups);
    };

    const toggleSchema = schemaGroup => ind => {
        let updatedGroup = [...schemaGroups[schemaGroup]];
        updatedGroup[ind] = {
            ...updatedGroup[ind],
            checked: !updatedGroup[ind].checked,
        };
        const updatedSchemaGroups = {
            ...schemaGroups,
            [schemaGroup]: updatedGroup,
        };
        setSchemaGroups(updatedSchemaGroups);
        propagateCheckedSchemas(updatedSchemaGroups);
    };

    return (
        <div className={s.container}>
            {loading && <CircularLoader />}
            {schemaGroups && (
                <>
                    <ButtonStrip>
                        <Button onClick={selectGenericHandler(true)}>
                            {i18n.t('Select All')}
                        </Button>
                        <Button onClick={selectGenericHandler(false)}>
                            {i18n.t('Select None')}
                        </Button>
                    </ButtonStrip>

                    <div className={s.formControl}>
                        {schemaGroupOrder.map(groupKey => {
                            const label = schemaGroupLabels[groupKey];

                            return (
                                <SchemaGroup
                                    key={label}
                                    label={label}
                                    schemas={schemaGroups[groupKey]}
                                    toggleSchema={toggleSchema(groupKey)}
                                />
                            );
                        })}
                    </div>
                </>
            )}
        </div>
    );
};

export { Schemas };
