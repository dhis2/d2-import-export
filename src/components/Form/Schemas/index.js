import React from 'react'
import { api } from 'services'
import i18n from '@dhis2/d2-i18n'
import { Loading } from 'components'
import { Checkbox, RaisedButton } from 'material-ui'
import { FormGroup, FormControl, FormLabel } from '../material-ui'

import s from './styles.css'

function groupName(klass) {
    let group = klass.split('.')
    group.pop()

    if (!klass.includes('.dhis')) {
        group.pop()
    }

    return group[group.length - 1]
}

function schemaGroups(schemas) {
    const groups = {}

    schemas.forEach(s => {
        if (!groups[s.group]) {
            groups[s.group] = []
        }

        groups[s.group].push(s)
    })

    // check groups with 1 item and merge inside Other
    const groupsWith1Item = []
    const OTHER_GROUP_NAME = 'other'
    let otherGroup = []

    Object.entries(groups).forEach(([k, v]) => {
        if (v.length === 1) {
            groupsWith1Item.push(k)
            v[0]['group'] = OTHER_GROUP_NAME
            otherGroup = otherGroup.concat(v)
        }
    })
    groups[OTHER_GROUP_NAME] = otherGroup

    groupsWith1Item.forEach(k => {
        delete groups[k]
    })

    // sort group items by length desc
    // Object.entries(groups).forEach(([k, v]) => {
    //   groups[k] = groups[k].sort(
    //     (a, b) => a.displayName.length - b.displayName.length
    //   )
    // })

    return groups
}

function breakOnCamelCase(schemaName, name) {
    let temp = schemaName.substr(0, name.length).replace(/([A-Z]+)/g, ' $1')
    return temp[0].toUpperCase() + temp.substr(1)
}

function groupLabel(name, schemas) {
    const nameLC = name.toLowerCase()
    if (nameLC === 'oauth2' || nameLC === 'other') {
        return name
    }

    for (let i = 0; i < schemas.length; i += 1) {
        if (nameLC === schemas[i]['name'].toLowerCase()) {
            return schemas[i]['displayName']
        }
    }

    for (let i = 0; i < schemas.length; i += 1) {
        const schemaName = schemas[i]['name'].toLowerCase()
        if (schemaName.includes(nameLC) && schemaName.indexOf(nameLC) === 0) {
            return breakOnCamelCase(schemas[i]['name'], name)
        }
    }

    return name[0].toUpperCase() + name.substr(1)
}

const styles = {
    iconStyle: {
        marginRight: 0,
    },
}

function Group({ label, schemas, checked, onClick }) {
    return (
        <div className={s.group}>
            <FormLabel className={s.formLabel}>{label}</FormLabel>
            <FormGroup className={s.schema}>
                {schemas.map(s => (
                    <Checkbox
                        {...styles}
                        key={`chk-${s.collectionName}`}
                        value={s.name}
                        label={s.displayName}
                        checked={checked.includes(s.collectionName)}
                        onCheck={(evt, status) =>
                            onClick(s.collectionName, status)
                        }
                    />
                ))}
            </FormGroup>
        </div>
    )
}

function Controls({ onSelectAll, onSelectNone }) {
    return (
        <div className={s.controls}>
            <RaisedButton label={i18n.t('Select All')} onClick={onSelectAll} />
            <RaisedButton
                label={i18n.t('Select None')}
                onClick={onSelectNone}
            />
        </div>
    )
}

export default class Schemas extends React.Component {
    state = {
        loaded: false,
        checked: [],
        schemas: [],
    }

    async componentDidMount() {
        this.fetch()
    }

    onClick = (collectionName, status) => {
        const { checked } = this.state

        let updated
        if (status) {
            updated = checked.slice(0)
            updated.push(collectionName)
        } else {
            updated = checked.filter(k => k !== collectionName)
        }

        this.setState({ checked: updated }, () =>
            this.props.onChange(this.props.name, updated)
        )
    }

    async fetch() {
        try {
            const {
                data: { schemas },
            } = await api.get('schemas.json')

            this.setState(
                {
                    loaded: true,
                    checked: schemas.map(item => item.collectionName),
                    schemas: schemas
                        .map(item => ({
                            name: item.name,
                            klass: item.klass,
                            displayName: item.displayName,
                            collectionName: item.collectionName,
                            group: groupName(item.klass),
                        }))
                        .sort((a, b) =>
                            a.displayName.localeCompare(b.displayName)
                        ),
                },
                () => {
                    this.props.onChange(this.props.name, this.state.checked)
                }
            )
        } catch (e) {
            console.log('fetch Schemas failed')
            console.log(e)
        }
    }

    onSelectNone = () => {
        this.setState({ checked: [] }, () =>
            this.props.onChange(this.props.name, [])
        )
    }

    onSelectAll = () => {
        const { schemas } = this.state
        this.setState(
            {
                checked: schemas.map(item => item.collectionName),
            },
            () => this.props.onChange(this.props.name, this.state.checked)
        )
    }

    render() {
        const { loaded, checked, schemas } = this.state
        if (!loaded) {
            return <Loading />
        }

        if (schemas.length === 0) {
            return null
        }

        const groups = schemaGroups(schemas)
        const list = Object.keys(groups).sort((a, b) => a.localeCompare(b))

        return (
            <div className={s.container}>
                <Controls
                    onSelectAll={this.onSelectAll}
                    onSelectNone={this.onSelectNone}
                />
                <FormControl className={s.formControl}>
                    {list.map(k => (
                        <Group
                            key={`group-${k}`}
                            label={groupLabel(k, groups[k])}
                            schemas={groups[k]}
                            checked={checked}
                            onClick={this.onClick}
                        />
                    ))}
                </FormControl>
            </div>
        )
    }
}
