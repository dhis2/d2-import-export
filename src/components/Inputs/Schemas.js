import { useDispatch, useSelector } from 'react-redux'
import React, { useEffect } from 'react'
import propTypes from 'prop-types'

import { Controls } from './Schemas/Controls'
import { Loading } from '../Loading'
import { SchemaGroup } from './Schemas/SchemaGroup'
import { fetchSchemas } from '../../reducers/schemas/thunks'
import {
    getGroupLabels,
    getGroupOrder,
    getSchemaGroups,
} from '../../reducers/schemas/selectors'
import s from './Schemas.module.css'

const SCHEMAS_KEY = 'schemas'

export const Schemas = ({ excludeSchemas }) => {
    const dispatch = useDispatch()
    const schemasLoaded = useSelector(state => state.schemas.loaded)
    const schemasLoading = useSelector(state => state.schemas.loading)
    const schemaGroups = useSelector(getSchemaGroups)
    const schemaGroupLabels = useSelector(getGroupLabels)
    const schemaGroupOrder = useSelector(getGroupOrder)

    useEffect(() => {
        if (!schemasLoading && !schemasLoaded) {
            dispatch(fetchSchemas(excludeSchemas))
        }
    }, [schemasLoading, schemasLoaded, excludeSchemas, dispatch])

    if (!schemasLoaded || schemasLoading) return <Loading />

    return (
        <div className={s.container} data-test="input-schemas">
            <Controls schemaKey={SCHEMAS_KEY} />

            <div className={s.formControl}>
                {schemaGroupOrder.map(groupKey => {
                    const label = schemaGroupLabels[groupKey]

                    return (
                        <SchemaGroup
                            key={label}
                            label={label}
                            schemas={schemaGroups[groupKey]}
                            value="true"
                        />
                    )
                })}
            </div>
        </div>
    )
}

Schemas.propTypes = {
    excludeSchemas: propTypes.instanceOf(Set),
}

Schemas.defaultProps = {
    excludeSchemas: new Set([]),
}
