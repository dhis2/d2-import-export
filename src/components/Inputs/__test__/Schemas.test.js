import { useSelector } from 'react-redux'

jest.unmock('react-final-form')

import { Form } from 'react-final-form'
import { mount } from 'enzyme'
import React from 'react'
import toJson from 'enzyme-to-json'

import { Schemas } from '../Schemas'

jest.mock('react-redux', () => ({
    useSelector: jest.fn(selector => selector()),
    useDispatch: jest.fn(() => action => action),
}))

const createState = () => ({
    schemas: {
        loaded: true,
        loading: false,
        error: '',
        list: [],
    },
})

describe('Input component - Schemas', () => {
    let state

    useSelector.mockImplementation(selector => selector(state))

    beforeEach(() => {
        state = createState()
    })

    it('should render correctly', () => {
        state.schemas.list = [
            {
                name: 'dataElements',
                label: 'Data elements',
                group: 'dataElement',
            },
            {
                name: 'dataElementGroups',
                label: 'Data element groups',
                group: 'dataElement',
            },
        ]

        const component = mount(
            <Form onSubmit={() => null}>{() => <Schemas />}</Form>
        )

        expect(toJson(component)).toMatchSnapshot()
    })
})
