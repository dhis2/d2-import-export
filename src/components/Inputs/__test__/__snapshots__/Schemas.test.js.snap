// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - Schemas should render correctly 1`] = `
<ReactFinalForm
  onSubmit={[Function]}
>
  <Schemas
    excludeSchemas={Set {}}
  >
    <div
      className="container"
      data-test="input-schemas"
    >
      <Controls
        schemaKey="schemas"
      >
        <div
          className="controls"
        >
          <SelectAll
            schemaKey="schemas"
          >
            <ToggleAll
              className="selectAllButton"
              schemaKey="schemas"
              value={true}
            >
              <Button
                className="selectAllButton"
                onClick={[Function]}
                type="button"
              >
                <button
                  className="jsx-3218119646 selectAllButton"
                  onClick={[Function]}
                  type="button"
                >
                  Select All
                  <JSXStyle
                    id="3218119646"
                  />
                </button>
              </Button>
            </ToggleAll>
          </SelectAll>
          <SelectNone
            schemaKey="schemas"
          >
            <ToggleAll
              schemaKey="schemas"
              value={false}
            >
              <Button
                onClick={[Function]}
                type="button"
              >
                <button
                  className="jsx-3218119646 "
                  onClick={[Function]}
                  type="button"
                >
                  Select None
                  <JSXStyle
                    id="3218119646"
                  />
                </button>
              </Button>
            </ToggleAll>
          </SelectNone>
        </div>
      </Controls>
      <div
        className="formControl"
      >
        <SchemaGroup
          key="Data Element"
          label="Data Element"
          schemas={
            Array [
              Object {
                "group": "dataElement",
                "label": "Data elements",
                "name": "dataElements",
              },
              Object {
                "group": "dataElement",
                "label": "Data element groups",
                "name": "dataElementGroups",
              },
            ]
          }
          value="true"
        >
          <div
            className="group"
          >
            <span
              className="formLabel"
            >
              Data Element
            </span>
            <div
              className="schema"
            >
              <div
                key="dataElements"
              >
                <Checkbox
                  format={[Function]}
                  initialValue="true"
                  label="Data elements"
                  name="schemas.dataElements"
                  parse={[Function]}
                >
                  <div>
                    <Checkbox
                      checked={true}
                      label="Data elements"
                      name="schemas.dataElements"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="checkbox"
                      value="irrelevant"
                    >
                      <label
                        className="jsx-1861485737 base"
                      >
                        <Input
                          checked={true}
                          name="schemas.dataElements"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          value="irrelevant"
                        >
                          <div
                            className="jsx-3255790221"
                          >
                            <input
                              checked={true}
                              className="jsx-3255790221"
                              name="schemas.dataElements"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              type="checkbox"
                              value="irrelevant"
                            />
                            <JSXStyle
                              id="3255790221"
                            />
                          </div>
                        </Input>
                        <Icon
                          checked={true}
                        >
                          <div
                            className="jsx-3164736166 "
                          >
                            <Checked
                              className="jsx-63432708 checked"
                            >
                              <svg
                                className="jsx-63432708 checked"
                                height="48"
                                viewBox="0 0 48 48"
                                width="48"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M38 6H10c-2.21 0-4 1.79-4 4v28c0 2.21 1.79 4 4 4h28c2.21 0 4-1.79 4-4V10c0-2.21-1.79-4-4-4zM20 34L10 24l2.83-2.83L20 28.34l15.17-15.17L38 16 20 34z"
                                />
                              </svg>
                            </Checked>
                            <JSXStyle
                              id="63432708"
                            />
                            <JSXStyle
                              dynamic={
                                Array [
                                  "#147cd7",
                                ]
                              }
                              id="1369204040"
                            />
                          </div>
                        </Icon>
                        <Label>
                          <span
                            className="jsx-583817354 label"
                          >
                            Data elements
                            <JSXStyle
                              dynamic={
                                Array [
                                  "#212934",
                                  "4px",
                                  "#6e7a8a",
                                ]
                              }
                              id="1864897066"
                            />
                          </span>
                        </Label>
                        <JSXStyle
                          dynamic={
                            Array [
                              "#6e7a8a",
                            ]
                          }
                          id="261820537"
                        />
                      </label>
                    </Checkbox>
                  </div>
                </Checkbox>
              </div>
              <div
                key="dataElementGroups"
              >
                <Checkbox
                  format={[Function]}
                  initialValue="true"
                  label="Data element groups"
                  name="schemas.dataElementGroups"
                  parse={[Function]}
                >
                  <div>
                    <Checkbox
                      checked={true}
                      label="Data element groups"
                      name="schemas.dataElementGroups"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="checkbox"
                      value="irrelevant"
                    >
                      <label
                        className="jsx-1861485737 base"
                      >
                        <Input
                          checked={true}
                          name="schemas.dataElementGroups"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          value="irrelevant"
                        >
                          <div
                            className="jsx-3255790221"
                          >
                            <input
                              checked={true}
                              className="jsx-3255790221"
                              name="schemas.dataElementGroups"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              type="checkbox"
                              value="irrelevant"
                            />
                            <JSXStyle
                              id="3255790221"
                            />
                          </div>
                        </Input>
                        <Icon
                          checked={true}
                        >
                          <div
                            className="jsx-3164736166 "
                          >
                            <Checked
                              className="jsx-63432708 checked"
                            >
                              <svg
                                className="jsx-63432708 checked"
                                height="48"
                                viewBox="0 0 48 48"
                                width="48"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M38 6H10c-2.21 0-4 1.79-4 4v28c0 2.21 1.79 4 4 4h28c2.21 0 4-1.79 4-4V10c0-2.21-1.79-4-4-4zM20 34L10 24l2.83-2.83L20 28.34l15.17-15.17L38 16 20 34z"
                                />
                              </svg>
                            </Checked>
                            <JSXStyle
                              id="63432708"
                            />
                            <JSXStyle
                              dynamic={
                                Array [
                                  "#147cd7",
                                ]
                              }
                              id="1369204040"
                            />
                          </div>
                        </Icon>
                        <Label>
                          <span
                            className="jsx-583817354 label"
                          >
                            Data element groups
                            <JSXStyle
                              dynamic={
                                Array [
                                  "#212934",
                                  "4px",
                                  "#6e7a8a",
                                ]
                              }
                              id="1864897066"
                            />
                          </span>
                        </Label>
                        <JSXStyle
                          dynamic={
                            Array [
                              "#6e7a8a",
                            ]
                          }
                          id="261820537"
                        />
                      </label>
                    </Checkbox>
                  </div>
                </Checkbox>
              </div>
            </div>
          </div>
        </SchemaGroup>
        <SchemaGroup
          key="other"
          label="other"
          schemas={Array []}
          value="true"
        >
          <div
            className="group"
          >
            <span
              className="formLabel"
            >
              other
            </span>
            <div
              className="schema"
            />
          </div>
        </SchemaGroup>
      </div>
    </div>
  </Schemas>
</ReactFinalForm>
`;
